features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_MCU_NRF51822/source/nRF5xGattServer.cpp

```diff
@@ -83,6 +84,7 @@ ble_error_t nRF5xGattServer::addService(GattService &service)
         for (uint8_t j = 0; j < p_char->getDescriptorCount(); j++) {
             GattAttribute *p_desc = p_char->getDescriptor(j);
             if (p_desc->getUUID() == BLE_UUID_DESCRIPTOR_CHAR_USER_DESC) {
+                description_descriptor = p_desc;
                 userDescriptionDescriptorValuePtr = p_desc->getValuePtr();
                 userDescriptionDescriptorValueLen = p_desc->getLength();
```

Comment:
Why It is expected that tere could be more than one descriptor per characteristic at input? Shouldn't this be an error?

Reply:
The Nordic code for the GattServer can be improved in many ways; raising an error if there is multiple user description descriptors may be one of them but it is not the purpose of this PR.

Reply:
Right. I Just spotted this at first sight.